packages/edgeidentity/ios/src/RCTAEPEdgeIdentityDataBridge.m

```diff
@@ -0,0 +1,122 @@
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+#import "RCTAEPEdgeIdentityDataBridge.h"
+
+@implementation RCTAEPEdgeIdentityDataBridge
+
+static NSString* const ID_KEY = @"id";
+static NSString* const IS_PRIMARY_KEY = @"primary";
+static NSString* const AUTH_STATE_KEY = @"authenticatedState";
+static NSString* const ITEM_KEY = @"items";
+static NSString* const AUTHENTICATED = @"authenticated";
+static NSString* const LOGGED_OUT = @"loggedOut";
+static NSString* const AMBIGUOUS = @"ambiguous";
+
++ (NSDictionary *)dictionaryFromIdentityMap: (nullable AEPIdentityMap *) idmap {
+    NSMutableDictionary *mapDict = [NSMutableDictionary dictionary];
+    
+    if (!idmap){
+        return mapDict;
+    }
+    
+        for (NSString *namespace in idmap.namespaces) {
+            NSArray* items = [idmap getItemsWithNamespace:namespace];
+           
+            NSMutableArray *mapArray = [NSMutableArray array];
+            
+            for (AEPIdentityItem *item in items){
+              NSMutableDictionary *itemdict = [NSMutableDictionary dictionary];
+              itemdict[IS_PRIMARY_KEY] = @(item.primary);
+              itemdict[AUTH_STATE_KEY] = [RCTAEPEdgeIdentityDataBridge stringFromAuthState:(item.authenticatedState)];
+              itemdict[ID_KEY] = item.id ;
+                
+              [mapArray addObject:itemdict];
+            }
+            if (mapArray.count !=0) {
+                [mapDict setObject:mapArray forKey:namespace];
+            }
+        }
+      
+        return mapDict;
+    }
+
++ (nonnull AEPIdentityMap  *)dictionaryToIdentityMap: (nonnull NSDictionary *) dict {
+    AEPIdentityMap *identityMap = [[AEPIdentityMap alloc] init];
+    NSDictionary *itemsMap = [[dict objectForKey:ITEM_KEY] isKindOfClass:[NSDictionary class]] ? [dict objectForKey:ITEM_KEY] : nil;
+    
+    if (!itemsMap){
+        return identityMap;
+    }
+    
+    NSArray <NSString*>* namespaces = [itemsMap allKeys];
+   
+    for (NSString *namespace in namespaces){
+        NSArray* items = [itemsMap objectForKey:namespace];
+        for (NSDictionary *itemMap in items){
+            AEPIdentityItem *item = [RCTAEPEdgeIdentityDataBridge dictionaryToIdentityItem:itemMap];
+            
+            if (item){
+                [identityMap addItem:item withNamespace:namespace];
+            }
+        }
+    }
+    
+    return identityMap;
+  }
+
++ (AEPIdentityItem *)dictionaryToIdentityItem: (nonnull NSDictionary *) dict {
```

Comment:
Was your intention to have this dictionaryToIdentityItem(nullable NSDictionary*) dict and check for nil bellow?

Reply:
Should I keep this as nonnull and don't need to check if dict is nil?

Reply:
You can keep the check inside and update the API signature to nullable.

Reply:
updated. thanks.